### ISSUE TITLE
[Bug] Bosconian and Galaga have slowdowns during explosions [sf#92]

### ISSUE THREAD

**Issue by openMSX-import** [openMSX/openMSX#393]
**Reported by manuelbi on 2003-05-17 15:05 UTC**
As said in the Summary: there are slowdowns during
explosions. Here's what Maarten said about it (testing
with Galaga):

he slowdown can be seen most clearly when the player
ship explodes, but also during gameplay if there is a
lot of action (especially explosions) going on.

I was unable to find the cause. In this mail I want to
list what I tried; 
might be useful in the future.

I thought the slowdown might connected to the sprites.
I tried to return fixed values from
SpriteChecker::readStatus, this had no influence. I
checked when readStatus is called, this is always on
line 235. In NTSC, display ends on line 234, so line
235 is pretty early in the interrupt routine; nothing
strange going on there. Sprites are sometimes slow in
SDLGL; however CPU usage is very low and the slowdown
also happens in SDLHi, so this is not the cause either.

Another theory is that the slowdown is connected to the
PSG. When something explodes, an explosion sound is
played. However, when disabling sound (F11), the
slowdown remains. Just to be sure I also disabled
updateBuffer in AY8910, that didn't make a difference. 

In theory, the Z80 timing could be wrong, but I don't
think this is likely, so I didn't check it.

In general, a slowdown can be caused by two things:
- timing inside emulated MSX is off
- realtime sync is off
Since none of the &amp;quot;inside&amp;quot; problems I could think of
seems to cause the slowdown, the realtime sync is what
remains. Maybe we should make an FPS counter, to see
whether openMSX really runs at 60Hz. We had some
problems with the SDL timers before, so it is worth it
to check this.

**Comment by openMSX-import** [comment]
**Commented by hmholschuh on 2006-01-27 04:28 UTC**
Logged In: YES 
user_id=1010847

I remember playing bosconian in a Brazilian Expert 1.1 MSX-1
(PAL-M Z80A MSX-1 64kB RAM -- timings are close to NTSC, not
to European PAL), and it *did* have slowdowns, although I am
not sure if they are of the same sort as those you are
experiencing.

So maybe the emulation is too perfect :P

**Comment by openMSX-import** [comment]
**Commented by m9710797 on 2006-09-12 18:41 UTC**
Logged In: YES 
user_id=356949

I did some more test on this:

\- i find slowdowns easiest to reproduce when there are a
lot of enemies on the screen and the player dies when there
are still enemies 'in flight'. Those enemies will return to
their place somewhat slower.
- during slowdown i didn't see any increase in CPU usage of
openmsx (remained constant at about 20%)
- slowdowns are easier to reproduce with NTSC then as PAL
- if i boost the Z80 clock to 7Mhz there are no slowdowns

All this only confirms hmholschuh's post. There might be
just too much work for a 3.5Mhz Z80 in one VDP interval.

So I believe this bug in invalid, but i'm still leaving it
open, i want to hear some other opinions first.

**Comment by openMSX-import** [comment]
**Updated by m9710797 on 2006-09-12 18:41 UTC**
- **priority**: 5 --> 3

**Comment by openMSX-import** [comment]
**Commented by m9710797 on 2006-09-12 19:12 UTC**
Logged In: YES 
user_id=356949

Hmm, i tried it on a real NMS8280 and turboR GT and there i
don't notice any slowdowns...

**Comment by openMSX-import** [comment]
**Commented by manuelbi on 2009-06-05 21:38 UTC**
The slowdown can be tested when the 3 aliens come down. Fly in the rain of bullets they send out and you'll die. You see the 3 aliens slowing down a lot, which doesn't happen on a real machine with Galaga.

I've retested this thoroughly with r9983 and that fix in IRQ handling seems to fix this problem as well!
Closing... (thanks for fixing, Wouter!) (After 6 years...!)
